test/wallet-importname-test.js

```diff
@@ -0,0 +1,226 @@
+    await mineBlocks(1);
+
+    // Send 3 bids in one block, confirm
+    await alice.sendBid(name, 100004, 200004);
+    await alice.sendBid(name, 100005, 200005);
+    await alice.sendBid(name, 100006, 200006);
+    await mineBlocks(1);
+
+    // One block with some bids for another name
+    await alice.sendBid(wrongName, 100007, 200007);
+    await alice.sendBid(wrongName, 100008, 200008);
+    await mineBlocks(1);
+
+    // Still in the bidding phase with one block left
+    const ns = await node.chain.db.getNameStateByName(name);
+    assert(ns.isBidding(node.chain.height + 1, network));
+
+    await wdb.rescan(0);
+
+    // Alice is tracking auction, Bob is not
+    const aliceBids = await alice.getBidsByName(name);
+    assert.strictEqual(aliceBids.length, 6);
+    for (const bid of aliceBids)
+      assert(bid.own);
+
+    const bobBids = await bob.getBidsByName(name);
+    assert.strictEqual(bobBids.length, 0);
+  });
+
+  it('should import name into Bob\'s wallet', async () => {
+    await bob.importName(name);
+
+    // Bob's wallet still has no name data for this name
+    const ns1 = await bob.getNameStateByName(name);
+    assert(ns1 === null);
+
+    // The WalletDB knows Alice & Bob are watching this name
+    const map1 = await wdb.getNameMap(nameHash);
+    assert(map1.wids.has(alice.wid));
+    assert(map1.wids.has(bob.wid));
+
+    // Only Alice is watching the other name
+    const map2 = await wdb.getNameMap(wrongNameHash);
+    assert(map1.wids.has(alice.wid));
+    assert(!map2.wids.has(bob.wid));
+  });
+
+  it('should not track bids for imported name before OPEN', async () => {
+    // Rescan covers bidding phase but does not include OPEN transaction
+    await wdb.rescan(6);
+
+    // Alice is tracking auction, Bob is not
+    const aliceBids = await alice.getBidsByName(name);
+    assert.strictEqual(aliceBids.length, 6);
+    for (const bid of aliceBids)
+      assert(bid.own);
+
+    const bobBids = await bob.getBidsByName(name);
+    assert.strictEqual(bobBids.length, 0);
+  });
+
+  it('should start tracking bids after seeing OPEN', async () => {
+    await wdb.rescan(0);
+
+    // After a sufficient rescan, Bob now has all auction data
+    const ns2 = await bob.getNameStateByName(name);
+    assert(ns2);
+
+    // ...even though he hasn't placed a bid yet
+    const bobBids = await bob.getBidsByName(name);
+    assert.strictEqual(bobBids.length, 6);
+    for (const bid of bobBids)
+      assert(!bid.own);
+
+    // Sanity check: only the "right" name was imported
+    const ns3 = await bob.getNameStateByName(wrongName);
+    assert(ns3 === null);
+    const ns4 = await alice.getNameStateByName(wrongName);
+    assert(ns4);
+  });
+
+  describe('rpc importname', function() {
+    it('should not have name data in Charlie\'s wallet', async () => {
+      const ns1 = await charlie.getNameStateByName(name);
+      assert(ns1 === null);
+    });
+
+    it('should rpc importname with rescan', async () => {
```

Comment:
nit: "should import name with rescan." The top-level describe makes "should rpc importname with rescan" redundant. Also the latter does read very well to me.